### DISCUSSION TITLE
Recursive `messages` and `attributes` methods with payload param

### DISCUSSION THREAD

**Discussion by RazorMeister** [spatie/laravel-data#181] [Ideas]
I think that functionality for custom validation messages and attributes should work similar to current `rules` method (recursive and with `payload` parameter). 

For example in my project I have dynamic address field attribute names depending on provided in request country id. I have a field - `zipCode` which is translated as `zip code` for USA and `postal code` for other countries.

**Comment by rubenvanassche** [comment]
I would accept a PR adding this feature.
